docs/index.bs

```diff
@@ -3288,29 +3288,44 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
-                  1. Resolve |preloadResponse| with |preloadResponseObject|.
-          1. [=If aborted=], then:
-              1. Let |deserializedError| be the result of [=deserialize a serialized abort reason=] given null and |workerRealm|.
-              1. [=fetch controller/Abort=] |preloadFetchController| with |deserializedError|.
+      1. Let |responseForAutoPreload| be null.
+      1. If |request| is a [=navigation request=], |request|'s [=request/method=] is \`<code>GET</code>\`, |registration|'s [=active worker=]'s [=set of event types to handle=] [=set/contains=] <code>fetch</code>, and |registration|'s [=active worker=]'s [=all fetch listeners are empty flag=] is not set then:
+          1. If |registration|'s [=navigation preload enabled flag=] is set then: 
+
+              Note: If the above is true except |registration|'s [=active worker=]'s <a>set of event types to handle</a> **does not** contain <code>fetch</code>, then the user agent may wish to show a console warning, as the developer's intent isn't clear.
+
+              1. Let |preloadRequest| be the result of [=request/cloning=] the request |request|.
+              1. Let |preloadRequestHeaders| be |preloadRequest|'s [=request/header list=].
+              1. Let |preloadResponseObject| be a new {{Response}} object associated with a new {{Headers}} object whose [=guard=] is "`immutable`".
+              1. [=header list/Append=] to |preloadRequestHeaders| a new [=header=] whose [=header/name=] is \`<code>Service-Worker-Navigation-Preload</code>\` and [=header/value=] is |registration|'s [=navigation preload header value=].
+              1. Set |preloadRequest|'s [=service-workers mode=] to "`none`".
+              1. Let |preloadFetchController| be null.
+              1. Run the following substeps [=in parallel=], but [=abort when=] |fetchController|'s [=fetch controller/state=] is "<code>terminated</code>" or "<code>aborted</code>":
+                  1. Set |preloadFetchController| to the result of [=Fetch|fetching=] |preloadRequest|.
+
+                      To [=fetch/processResponse=] for |navigationPreloadResponse|, run these substeps:
+
+                      1. If |navigationPreloadResponse|'s [=response/type=] is "`error`", reject |preloadResponse| with a `TypeError` and terminate these substeps.
+                      1. Associate |preloadResponseObject| with |navigationPreloadResponse|.
+                      1. Resolve |preloadResponse| with |preloadResponseObject|.
+              1. [=If aborted=], then:
+                  1. Let |deserializedError| be the result of [=deserialize a serialized abort reason=] given null and |workerRealm|.
+                  1. [=fetch controller/Abort=] |preloadFetchController| with |deserializedError|.
+          1. Else if |timingInfo|'s [=service worker timing info/worker matched router source=] is not "{{RouterSourceEnum/fetch-event}}", a user agent may run the following substeps:
```

Comment:
Updated not to introduce |source| here, use |timingInfo|'s [=service worker timing info/worker matched router source=] instead.

Reply:
Thanks.

Reply:
We frame the autopreload is an optional optimization that the browser can apply at its choosing. So we want to use the static routing API as an opt-out mechanism. If we only invoke the autopreload only when the matched router source is `fetch-event`, that is more like an opt-in, not opt-out.

Reply:
Thank you! That sounds reasonable. Added `Assert: [=service worker timing info/worker matched router source=] is null.`.